pricegraph/src/orderbook.rs

```diff
@@ -233,7 +241,7 @@ impl Orderbook {
             assert!(spread > 0.0, "invalid spread");
         }
 
-        if !self.is_token_pair_valid(pair) {
+        if !self.is_token_pair_valid(pair) || pair.buy == pair.sell {
```

Comment:
This is the infinite loop.

Reply:
Good catch!

Reply:
nit: I would have this be a `is_market_valid` method that is `is_token_pair_valid && buy != sell`.